clients/cli/cmd/internal/pull.go

```diff
@@ -155,47 +157,146 @@ func (target *Target) DownloadAndWriteToFile(client *phrase.APIClient, localeFil
 		localVarOptionals.Tag = optional.EmptyString()
 	}
 
-	if Debug {
-		fmt.Fprintln(os.Stderr, "Target file pattern:", target.File)
-		fmt.Fprintln(os.Stderr, "Actual file path", localeFile.Path)
-		fmt.Fprintln(os.Stderr, "LocaleID", localeFile.ID)
-		fmt.Fprintln(os.Stderr, "ProjectID", target.ProjectID)
-		fmt.Fprintln(os.Stderr, "FileFormat", localVarOptionals.FileFormat)
-		fmt.Fprintln(os.Stderr, "ConvertEmoji", localVarOptionals.ConvertEmoji)
-		fmt.Fprintln(os.Stderr, "IncludeEmptyTranslations", localVarOptionals.IncludeEmptyTranslations)
-		fmt.Fprintln(os.Stderr, "KeepNotranslateTags", localVarOptionals.KeepNotranslateTags)
-		fmt.Fprintln(os.Stderr, "Tags", localVarOptionals.Tags)
-		fmt.Fprintln(os.Stderr, "Branch", localVarOptionals.Branch)
-		fmt.Fprintln(os.Stderr, "FormatOptions", localVarOptionals.FormatOptions)
-	}
-
-	file, response, err := client.LocalesApi.LocaleDownload(Auth, target.ProjectID, localeFile.ID, &localVarOptionals)
+	debugFprintln("Target file pattern:", target.File)
+	debugFprintln("Actual file path", localeFile.Path)
+	debugFprintln("LocaleID", localeFile.ID)
+	debugFprintln("ProjectID", target.ProjectID)
+	debugFprintln("FileFormat", localVarOptionals.FileFormat)
+	debugFprintln("ConvertEmoji", localVarOptionals.ConvertEmoji)
+	debugFprintln("IncludeEmptyTranslations", localVarOptionals.IncludeEmptyTranslations)
+	debugFprintln("KeepNotranslateTags", localVarOptionals.KeepNotranslateTags)
+	debugFprintln("Tags", localVarOptionals.Tags)
+	debugFprintln("Branch", localVarOptionals.Branch)
+	debugFprintln("FormatOptions", localVarOptionals.FormatOptions)
+
+	if async {
+		return target.downloadAsynchronously(client, localeFile, localVarOptionals)
+	} else {
+		return target.downloadSynchronously(client, localeFile, localVarOptionals)
+	}
+}
+
+func (target *Target) downloadAsynchronously(client *phrase.APIClient, localeFile *LocaleFile, downloadOpts phrase.LocaleDownloadOpts) error {
+	localeDownloadCreateParams := asyncDownloadParams(downloadOpts)
+
+	localVarOptionals := phrase.LocaleDownloadCreateOpts{}
+	debugFprintln("Initiating async download...")
+	asyncDownload, _, err := client.LocaleDownloadsApi.LocaleDownloadCreate(Auth, target.ProjectID, localeFile.ID, localeDownloadCreateParams, &localVarOptionals)
+	if err != nil {
+		return err
+	}
+
+	for asyncDownload.Status == "processing" {
```

Comment:
Thinking whether also should introduce a total timeout 🤔 Like 30 minutes for example